### ISSUE TITLE
About error handling

### ISSUE THREAD

**Issue by FeiZhu** [FeiZhu/Physika#51]
We don't have to exit the program every time we're faced with invalid input. For instance, in LightManager we only maintain at most 8 lights. But we don't have to exit the program if the user is about to add the ninth light. Instead, the better solution is to warn the user about this and ignore the ninth light. Another example, in SurfaceMeshRender we provide methods to render specific faces with custom color. The number of input colors is expected to be the same as the number of faces to render. But if the user input less colors than the faces, instead of exiting the program we should render the unspecified faces with default color.

In short, exiting the program is fatal, we should be extremely certain when we decide to exit the program.
